draco/analysis/flagging.py

```diff
@@ -581,10 +581,7 @@ def process(self, sensitivity):
             finalmask[:, madtimes] = madmask[:, madtimes]
 
         # Collect all parts of the mask onto rank 1 and then broadcast to all ranks
-        finalmask = mpiarray.MPIArray.wrap(finalmask[np.newaxis, ...], 1).redistribute(
-            axis=0
-        )
-        finalmask = sensitivity.comm.bcast(finalmask[0], root=0)
+        finalmask = mpiarray.MPIArray.wrap(finalmask, 0).allgather()
```

Comment:
It took me a while to make sense of the old snippet here even knowing what behaviour the new line is recreating...